scripts/init.sh

```diff
@@ -2,11 +2,17 @@
 
 set -e
 
-echo "*** Initializing WASM build environment"
+echo "*** Initializing build environment"
 
-if [ -z $CI_PROJECT_NAME ] ; then
-   rustup update nightly-2022-04-13
-   rustup update stable
+# No sudo during docker build
+if [ -f /.dockerenv ]; then
+   apt-get update && \
+   apt-get install -y build-essential clang curl libssl-dev protobuf-compiler
+else
+   sudo apt-get update && \
+   sudo apt-get install -y build-essential clang curl libssl-dev protobuf-compiler
 fi
 
-rustup target add wasm32-unknown-unknown --toolchain nightly-2022-04-13
+curl https://sh.rustup.rs -sSf | sh -s -- -y && \
```

Comment:
The newest rust compiler version is taken, right? not our specified from the toolchain at this point.

Reply:
It installs rustup and cargo. Then the script executes `rustup show` to install the toolchain and targets specified in the "rust-toolchain" file

Reply:
It installs rustup and cargo. Then the script executes `rustup show` to install the toolchain and targets specified in the "rust-toolchain" file